classes/plugins/ImportExportPlugin.inc.php

```diff
@@ -181,6 +186,208 @@ function displayXMLValidationErrors($errors, $xml) {
 		throw new Exception(__('plugins.importexport.common.error.validation'));
 	}
 
+	/**
+	 * Set the deployment that processes import/export operations
+	 */
+	public function setDeployment($deployment) {
+		$this->_childDeployment = $deployment;
+	}
+
+	/**
+	 * Get the deployment that processes import/export operations
+	 * @return PKPImportExportDeployment
+	 */
+	public function getDeployment() {
+		return $this->_childDeployment;
+	}
+
+	/**
+	 * Get the submissions and proceed to the export
+	 * @param $submissionIds array Array of submissions to export
+	 * @param $deployment PKPNativeImportExportDeployment
+	 * @param $opts array
+	 */
+	function getExportSubmissionsDeployment($submissionIds, &$deployment, $opts = array()) {
+		$filter = $this->getExportFilter('exportSubmissions');
+
+		$submissions = array();
+		foreach ($submissionIds as $submissionId) {
+			/** @var $submissionService APP\Services\SubmissionService */
+			$submissionService = Services::get('submission');
+			$submission = $submissionService->get($submissionId);
```

Comment:
Should this be added to `get` service method?

Reply:
We used to have it in the DAO getters and I do worry that we lost the capability to concisely ensure integrity when we moved to the service class. @NateWr, I know we've talked about this before but I forget the outcome -- do you have an equivalent mechanism in mind for service classes?

Reply:
I can't find the surrounding code for this in the modified files of this PR, so I can't tell where `$submissionIds` come from. If it is from a `getIds` call, the restriction should be done there:

```php
$submissionIds = Services::get('submission')->getIds(['contextId' => $contextId]);
```

If the `$submissionIds` come from user input or an unfiltered source, the check should be done elsewhere:

```
if ($submission->getData('contextId') !== $contextId) {
  continue;
}
```

I'm not in favor of adding context restrictions to methods like this because the service classes are designed to live "above" the level of the context. Only the handlers know what the appropriate context for a call to the service class is and shouldn't assume that the request contexts always match (which is how much of the previous code worked, which prevented it from being used at, say, the admin level, which will be important for processing job queues and things).

I think that if we need a performant way to test a set of `$submissionIds` against a context, we should write a method to do that.

Reply:
Fixed